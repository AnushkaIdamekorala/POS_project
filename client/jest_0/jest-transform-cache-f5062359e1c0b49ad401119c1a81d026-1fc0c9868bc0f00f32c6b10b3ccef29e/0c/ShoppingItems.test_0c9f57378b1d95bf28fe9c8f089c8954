14e3873f2acfa7fcca3ac427e171144c
"use strict";

var _interopRequireDefault = require("/home/anushkai/Desktop/Training/code/redux/fullStack/client/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("/home/anushkai/Desktop/Training/code/redux/fullStack/client/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireWildcard");

var _react = _interopRequireDefault(require("react"));

var _enzyme = _interopRequireWildcard(require("enzyme"));

var _enzymeAdapterReact = _interopRequireDefault(require("enzyme-adapter-react-16"));

var _ShoppingItems = _interopRequireDefault(require("../components/ShoppingItems"));

var _reduxThunk = _interopRequireDefault(require("redux-thunk"));

var _reactRedux = require("react-redux");

var _sinon = _interopRequireDefault(require("sinon"));

var _reduxMockStore = _interopRequireDefault(require("redux-mock-store"));

var _jsxFileName = "/home/anushkai/Desktop/Training/code/redux/fullStack/client/src/client_test_/ShoppingItems.test.js";
const middlewares = [_reduxThunk.default];

_enzyme.default.configure({
  adapter: new _enzymeAdapterReact.default()
});

describe("components", () => {
  describe("Login component", () => {
    const mockStore = (0, _reduxMockStore.default)(middlewares);
    let store;

    const loginUser = _sinon.default.spy();

    const userCarts = _sinon.default.spy();

    const initialState = {
      item: {
        items: []
      }
    };
    beforeEach(() => {
      store = mockStore(initialState);
    });
    it("should render self and subcomponents", () => {
      /* const props = {
        login: jest.fn()
      }; */
      const wrapper = (0, _enzyme.mount)(_react.default.createElement(_reactRedux.Provider, {
        store: store,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 36
        },
        __self: void 0
      }, _react.default.createElement(_ShoppingItems.default, {
        __source: {
          fileName: _jsxFileName,
          lineNumber: 37
        },
        __self: void 0
      })));
      expect(wrapper).toMatchSnapshot();
    });
    /* it("simulate button action", () => {
      const wrapper = mount(
        <Provider store={store}>
          <ShoppingItems />
        </Provider>
      );
      
    });*/
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlNob3BwaW5nSXRlbXMudGVzdC5qcyJdLCJuYW1lcyI6WyJtaWRkbGV3YXJlcyIsInRodW5rIiwiRW56eW1lIiwiY29uZmlndXJlIiwiYWRhcHRlciIsIkFkYXB0ZXIiLCJkZXNjcmliZSIsIm1vY2tTdG9yZSIsInN0b3JlIiwibG9naW5Vc2VyIiwic2lub24iLCJzcHkiLCJ1c2VyQ2FydHMiLCJpbml0aWFsU3RhdGUiLCJpdGVtIiwiaXRlbXMiLCJiZWZvcmVFYWNoIiwiaXQiLCJ3cmFwcGVyIiwiZXhwZWN0IiwidG9NYXRjaFNuYXBzaG90Il0sIm1hcHBpbmdzIjoiOzs7Ozs7QUFBQTs7QUFDQTs7QUFDQTs7QUFDQTs7QUFDQTs7QUFDQTs7QUFDQTs7QUFFQTs7O0FBRUEsTUFBTUEsV0FBVyxHQUFHLENBQUNDLG1CQUFELENBQXBCOztBQUVBQyxnQkFBT0MsU0FBUCxDQUFpQjtBQUFFQyxFQUFBQSxPQUFPLEVBQUUsSUFBSUMsMkJBQUo7QUFBWCxDQUFqQjs7QUFFQUMsUUFBUSxDQUFDLFlBQUQsRUFBZSxNQUFNO0FBQzNCQSxFQUFBQSxRQUFRLENBQUMsaUJBQUQsRUFBb0IsTUFBTTtBQUNoQyxVQUFNQyxTQUFTLEdBQUcsNkJBQWVQLFdBQWYsQ0FBbEI7QUFDQSxRQUFJUSxLQUFKOztBQUNBLFVBQU1DLFNBQVMsR0FBR0MsZUFBTUMsR0FBTixFQUFsQjs7QUFDQSxVQUFNQyxTQUFTLEdBQUdGLGVBQU1DLEdBQU4sRUFBbEI7O0FBRUEsVUFBTUUsWUFBWSxHQUFHO0FBQ25CQyxNQUFBQSxJQUFJLEVBQUU7QUFDSkMsUUFBQUEsS0FBSyxFQUFFO0FBREg7QUFEYSxLQUFyQjtBQU1BQyxJQUFBQSxVQUFVLENBQUMsTUFBTTtBQUNmUixNQUFBQSxLQUFLLEdBQUdELFNBQVMsQ0FBQ00sWUFBRCxDQUFqQjtBQUNELEtBRlMsQ0FBVjtBQUdBSSxJQUFBQSxFQUFFLENBQUMsc0NBQUQsRUFBeUMsTUFBTTtBQUMvQzs7O0FBR0EsWUFBTUMsT0FBTyxHQUFHLG1CQUNkLDZCQUFDLG9CQUFEO0FBQVUsUUFBQSxLQUFLLEVBQUVWLEtBQWpCO0FBQUE7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBLFNBQ0UsNkJBQUMsc0JBQUQ7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBO0FBQUEsUUFERixDQURjLENBQWhCO0FBS0FXLE1BQUFBLE1BQU0sQ0FBQ0QsT0FBRCxDQUFOLENBQWdCRSxlQUFoQjtBQUNELEtBVkMsQ0FBRjtBQVlBOzs7Ozs7OztBQVFELEdBbkNPLENBQVI7QUFvQ0QsQ0FyQ08sQ0FBUiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCBmcm9tIFwicmVhY3RcIjtcbmltcG9ydCBFbnp5bWUsIHsgc2hhbGxvdywgbW91bnQgfSBmcm9tIFwiZW56eW1lXCI7XG5pbXBvcnQgQWRhcHRlciBmcm9tIFwiZW56eW1lLWFkYXB0ZXItcmVhY3QtMTZcIjtcbmltcG9ydCBTaG9wcGluZ0l0ZW1zIGZyb20gXCIuLi9jb21wb25lbnRzL1Nob3BwaW5nSXRlbXNcIjtcbmltcG9ydCB0aHVuayBmcm9tIFwicmVkdXgtdGh1bmtcIjtcbmltcG9ydCB7IFByb3ZpZGVyIH0gZnJvbSBcInJlYWN0LXJlZHV4XCI7XG5pbXBvcnQgc2lub24gZnJvbSBcInNpbm9uXCI7XG5cbmltcG9ydCBjb25maWd1cmVTdG9yZSBmcm9tIFwicmVkdXgtbW9jay1zdG9yZVwiO1xuXG5jb25zdCBtaWRkbGV3YXJlcyA9IFt0aHVua107XG5cbkVuenltZS5jb25maWd1cmUoeyBhZGFwdGVyOiBuZXcgQWRhcHRlcigpIH0pO1xuXG5kZXNjcmliZShcImNvbXBvbmVudHNcIiwgKCkgPT4ge1xuICBkZXNjcmliZShcIkxvZ2luIGNvbXBvbmVudFwiLCAoKSA9PiB7XG4gICAgY29uc3QgbW9ja1N0b3JlID0gY29uZmlndXJlU3RvcmUobWlkZGxld2FyZXMpO1xuICAgIGxldCBzdG9yZTtcbiAgICBjb25zdCBsb2dpblVzZXIgPSBzaW5vbi5zcHkoKTtcbiAgICBjb25zdCB1c2VyQ2FydHMgPSBzaW5vbi5zcHkoKTtcblxuICAgIGNvbnN0IGluaXRpYWxTdGF0ZSA9IHtcbiAgICAgIGl0ZW06IHtcbiAgICAgICAgaXRlbXM6IFtdXG4gICAgICB9XG4gICAgfTtcblxuICAgIGJlZm9yZUVhY2goKCkgPT4ge1xuICAgICAgc3RvcmUgPSBtb2NrU3RvcmUoaW5pdGlhbFN0YXRlKTtcbiAgICB9KTtcbiAgICBpdChcInNob3VsZCByZW5kZXIgc2VsZiBhbmQgc3ViY29tcG9uZW50c1wiLCAoKSA9PiB7XG4gICAgICAvKiBjb25zdCBwcm9wcyA9IHtcbiAgICAgICAgbG9naW46IGplc3QuZm4oKVxuICAgICAgfTsgKi9cbiAgICAgIGNvbnN0IHdyYXBwZXIgPSBtb3VudChcbiAgICAgICAgPFByb3ZpZGVyIHN0b3JlPXtzdG9yZX0+XG4gICAgICAgICAgPFNob3BwaW5nSXRlbXMgLz5cbiAgICAgICAgPC9Qcm92aWRlcj5cbiAgICAgICk7XG4gICAgICBleHBlY3Qod3JhcHBlcikudG9NYXRjaFNuYXBzaG90KCk7XG4gICAgfSk7XG5cbiAgICAvKiBpdChcInNpbXVsYXRlIGJ1dHRvbiBhY3Rpb25cIiwgKCkgPT4ge1xuICAgICAgY29uc3Qgd3JhcHBlciA9IG1vdW50KFxuICAgICAgICA8UHJvdmlkZXIgc3RvcmU9e3N0b3JlfT5cbiAgICAgICAgICA8U2hvcHBpbmdJdGVtcyAvPlxuICAgICAgICA8L1Byb3ZpZGVyPlxuICAgICAgKTtcbiAgICAgIFxuICAgIH0pOyovXG4gIH0pO1xufSk7XG4iXX0=