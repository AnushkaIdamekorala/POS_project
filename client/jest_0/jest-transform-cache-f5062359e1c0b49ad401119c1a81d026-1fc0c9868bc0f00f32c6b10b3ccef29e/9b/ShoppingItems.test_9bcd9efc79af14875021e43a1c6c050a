2a7ceffdf45f7b9076c1725d08aa4a53
"use strict";

var _interopRequireDefault = require("/home/anushkai/Desktop/Training/code/redux/fullStack/client/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("/home/anushkai/Desktop/Training/code/redux/fullStack/client/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireWildcard");

var _react = _interopRequireDefault(require("react"));

var _enzyme = _interopRequireWildcard(require("enzyme"));

var _enzymeAdapterReact = _interopRequireDefault(require("enzyme-adapter-react-16"));

var _ShoppingItems = _interopRequireDefault(require("../components/ShoppingItems"));

var _reduxThunk = _interopRequireDefault(require("redux-thunk"));

var _reactRedux = require("react-redux");

var _sinon = _interopRequireDefault(require("sinon"));

var _reduxMockStore = _interopRequireDefault(require("redux-mock-store"));

var _jsxFileName = "/home/anushkai/Desktop/Training/code/redux/fullStack/client/src/client_test_/ShoppingItems.test.js";
const middlewares = [_reduxThunk.default];

_enzyme.default.configure({
  adapter: new _enzymeAdapterReact.default()
});

describe("components", () => {
  describe("Login component", () => {
    const mockStore = (0, _reduxMockStore.default)(middlewares);
    let store;

    const loginUser = _sinon.default.spy();

    const userCarts = _sinon.default.spy();

    const initialState = {
      item: {
        items: []
      },
      cart: {
        cartItems: []
      }
    };
    beforeEach(() => {
      store = mockStore(initialState);
    });
    it("should render self and subcomponents", () => {
      /* const props = {
        login: jest.fn()
      }; */
      const wrapper = (0, _enzyme.mount)(_react.default.createElement(_reactRedux.Provider, {
        store: store,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 39
        },
        __self: void 0
      }, _react.default.createElement(_ShoppingItems.default, {
        __source: {
          fileName: _jsxFileName,
          lineNumber: 40
        },
        __self: void 0
      })));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlNob3BwaW5nSXRlbXMudGVzdC5qcyJdLCJuYW1lcyI6WyJtaWRkbGV3YXJlcyIsInRodW5rIiwiRW56eW1lIiwiY29uZmlndXJlIiwiYWRhcHRlciIsIkFkYXB0ZXIiLCJkZXNjcmliZSIsIm1vY2tTdG9yZSIsInN0b3JlIiwibG9naW5Vc2VyIiwic2lub24iLCJzcHkiLCJ1c2VyQ2FydHMiLCJpbml0aWFsU3RhdGUiLCJpdGVtIiwiaXRlbXMiLCJjYXJ0IiwiY2FydEl0ZW1zIiwiYmVmb3JlRWFjaCIsIml0Iiwid3JhcHBlciIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCJdLCJtYXBwaW5ncyI6Ijs7Ozs7O0FBQUE7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBRUE7OztBQUVBLE1BQU1BLFdBQVcsR0FBRyxDQUFDQyxtQkFBRCxDQUFwQjs7QUFFQUMsZ0JBQU9DLFNBQVAsQ0FBaUI7QUFBRUMsRUFBQUEsT0FBTyxFQUFFLElBQUlDLDJCQUFKO0FBQVgsQ0FBakI7O0FBRUFDLFFBQVEsQ0FBQyxZQUFELEVBQWUsTUFBTTtBQUMzQkEsRUFBQUEsUUFBUSxDQUFDLGlCQUFELEVBQW9CLE1BQU07QUFDaEMsVUFBTUMsU0FBUyxHQUFHLDZCQUFlUCxXQUFmLENBQWxCO0FBQ0EsUUFBSVEsS0FBSjs7QUFDQSxVQUFNQyxTQUFTLEdBQUdDLGVBQU1DLEdBQU4sRUFBbEI7O0FBQ0EsVUFBTUMsU0FBUyxHQUFHRixlQUFNQyxHQUFOLEVBQWxCOztBQUVBLFVBQU1FLFlBQVksR0FBRztBQUNuQkMsTUFBQUEsSUFBSSxFQUFFO0FBQ0pDLFFBQUFBLEtBQUssRUFBRTtBQURILE9BRGE7QUFJbkJDLE1BQUFBLElBQUksRUFBRTtBQUNKQyxRQUFBQSxTQUFTLEVBQUU7QUFEUDtBQUphLEtBQXJCO0FBU0FDLElBQUFBLFVBQVUsQ0FBQyxNQUFNO0FBQ2ZWLE1BQUFBLEtBQUssR0FBR0QsU0FBUyxDQUFDTSxZQUFELENBQWpCO0FBQ0QsS0FGUyxDQUFWO0FBR0FNLElBQUFBLEVBQUUsQ0FBQyxzQ0FBRCxFQUF5QyxNQUFNO0FBQy9DOzs7QUFHQSxZQUFNQyxPQUFPLEdBQUcsbUJBQ2QsNkJBQUMsb0JBQUQ7QUFBVSxRQUFBLEtBQUssRUFBRVosS0FBakI7QUFBQTtBQUFBO0FBQUE7QUFBQTtBQUFBO0FBQUEsU0FDRSw2QkFBQyxzQkFBRDtBQUFBO0FBQUE7QUFBQTtBQUFBO0FBQUE7QUFBQSxRQURGLENBRGMsQ0FBaEI7QUFLQWEsTUFBQUEsTUFBTSxDQUFDRCxPQUFELENBQU4sQ0FBZ0JFLGVBQWhCO0FBQ0QsS0FWQyxDQUFGO0FBV0QsR0E3Qk8sQ0FBUjtBQThCRCxDQS9CTyxDQUFSIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gXCJyZWFjdFwiO1xuaW1wb3J0IEVuenltZSwgeyBzaGFsbG93LCBtb3VudCB9IGZyb20gXCJlbnp5bWVcIjtcbmltcG9ydCBBZGFwdGVyIGZyb20gXCJlbnp5bWUtYWRhcHRlci1yZWFjdC0xNlwiO1xuaW1wb3J0IFNob3BwaW5nSXRlbXMgZnJvbSBcIi4uL2NvbXBvbmVudHMvU2hvcHBpbmdJdGVtc1wiO1xuaW1wb3J0IHRodW5rIGZyb20gXCJyZWR1eC10aHVua1wiO1xuaW1wb3J0IHsgUHJvdmlkZXIgfSBmcm9tIFwicmVhY3QtcmVkdXhcIjtcbmltcG9ydCBzaW5vbiBmcm9tIFwic2lub25cIjtcblxuaW1wb3J0IGNvbmZpZ3VyZVN0b3JlIGZyb20gXCJyZWR1eC1tb2NrLXN0b3JlXCI7XG5cbmNvbnN0IG1pZGRsZXdhcmVzID0gW3RodW5rXTtcblxuRW56eW1lLmNvbmZpZ3VyZSh7IGFkYXB0ZXI6IG5ldyBBZGFwdGVyKCkgfSk7XG5cbmRlc2NyaWJlKFwiY29tcG9uZW50c1wiLCAoKSA9PiB7XG4gIGRlc2NyaWJlKFwiTG9naW4gY29tcG9uZW50XCIsICgpID0+IHtcbiAgICBjb25zdCBtb2NrU3RvcmUgPSBjb25maWd1cmVTdG9yZShtaWRkbGV3YXJlcyk7XG4gICAgbGV0IHN0b3JlO1xuICAgIGNvbnN0IGxvZ2luVXNlciA9IHNpbm9uLnNweSgpO1xuICAgIGNvbnN0IHVzZXJDYXJ0cyA9IHNpbm9uLnNweSgpO1xuXG4gICAgY29uc3QgaW5pdGlhbFN0YXRlID0ge1xuICAgICAgaXRlbToge1xuICAgICAgICBpdGVtczogW11cbiAgICAgIH0sXG4gICAgICBjYXJ0OiB7XG4gICAgICAgIGNhcnRJdGVtczogW11cbiAgICAgIH1cbiAgICB9O1xuXG4gICAgYmVmb3JlRWFjaCgoKSA9PiB7XG4gICAgICBzdG9yZSA9IG1vY2tTdG9yZShpbml0aWFsU3RhdGUpO1xuICAgIH0pO1xuICAgIGl0KFwic2hvdWxkIHJlbmRlciBzZWxmIGFuZCBzdWJjb21wb25lbnRzXCIsICgpID0+IHtcbiAgICAgIC8qIGNvbnN0IHByb3BzID0ge1xuICAgICAgICBsb2dpbjogamVzdC5mbigpXG4gICAgICB9OyAqL1xuICAgICAgY29uc3Qgd3JhcHBlciA9IG1vdW50KFxuICAgICAgICA8UHJvdmlkZXIgc3RvcmU9e3N0b3JlfT5cbiAgICAgICAgICA8U2hvcHBpbmdJdGVtcyAvPlxuICAgICAgICA8L1Byb3ZpZGVyPlxuICAgICAgKTtcbiAgICAgIGV4cGVjdCh3cmFwcGVyKS50b01hdGNoU25hcHNob3QoKTtcbiAgICB9KTtcbiAgfSk7XG59KTtcbiJdfQ==