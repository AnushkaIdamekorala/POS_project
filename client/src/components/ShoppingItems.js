import { Container, ListGroup, ListGroupItem, Button } from "reactstrap";
import { Transition, TransitionGroup } from "react-transition-group";
import React, { Component } from "react";
import { connect } from "react-redux";
import { getItems, deleteItem } from "../actions/itemActions";
import { addToCart } from "../actions/cartActions";

import PropTypes from "prop-types";

class ShoppingItems extends Component {
  componentDidMount() {
    this.props.getItems();
  }

  onDeleteClick = id => {
    this.props.deleteItem(id);
  };

  onAddClick = (id, name) => {
    let count = parseInt(document.getElementById(id).value, 10);
    this.props.addToCart(id, count);
    document.getElementById(name).className = "disabled";
  };

  render() {
    const { items } = this.props.item;
    const { cartItems } = this.props.cart;
    var cartUnwraps = cartItems.map(i => {
      return i.itm;
    });
    var mari = function(a, b) {
      if (
        a
          .filter(i => {
            return b === i._id;
          })
          .toString()
      ) {
        return "disabled";
      } else {
        return "";
      }
    };

    return (
      <React.Fragment>
        <div className="containerer">
          <div className="card shopping-cart">
            <div className="card-header bg-dark text-light">
              <span> </span>
              Here is what we got!!!
              <div className="clearfix" />
            </div>
            <div className="card-body">
              {/* PRODUCT */}
              {items.map(it => (
                <div
                  key={it._id}
                  id={it.name}
                  className={mari(cartUnwraps, it._id)}
                >
                  <div className="row">
                    <div className="col-12 col-sm-12 col-md-2 text-center">
                      <img
                        className="img-responsive"
                        src="https://picsum.photos/120/80"
                        alt={it.name}
                        width={120}
                        height={80}
                      />
                    </div>
                    <div className="col-12 text-sm-center col-sm-12 text-md-left col-md-6">
                      <h4 className="product-name">
                        <strong>{it.name}</strong>
                      </h4>
                      <h4>
                        <small>{it.description}</small>
                      </h4>
                    </div>

                    <div className="col-12 col-sm-12 text-sm-center col-md-4 text-md-right row">
                      <div
                        className="col-6 col-sm-6 col-md-6 text-md-right"
                        style={{ paddingTop: "5px" }}
                      >
                        <h5>
                          <strong>
                            ${Number.parseFloat(it.price).toFixed(2)}
                            <span className="text-muted">{" X"}</span>
                          </strong>
                        </h5>
                      </div>
                      <div className="col-4 col-sm-4 col-md-4 text-md-left">
                        <div className="quantity">
                          <input
                            id={it._id}
                            className="nextLevel"
                            type="number"
                            step={1}
                            defaultValue={1}
                            min={1}
                            title="Qty"
                            className="form-control form-control-sm"
                            size={4}
                          />
                          <br />
                          <small>
                            <em>upto {it.available} units </em>
                          </small>
                        </div>
                      </div>
                      <div className="col-2 col-sm-2 col-md-2 text-right">
                        <button
                          onClick={this.onAddClick.bind(this, it._id, it.name)}
                          type="button"
                          className="btn btn-outline-success btn-s"
                        >
                          <i class="fa fa-cart-plus" aria-hidden="true" />
                        </button>
                      </div>
                    </div>
                  </div>
                  <hr />
                </div>
              ))}
              <div className="float-right">
                <a className="btn btn-outline-secondary float-right">
                  Add Items
                </a>
              </div>
            </div>
            <div className="card-footer">
              <div className="coupon col-md-5 col-sm-5 no-padding-left float-left">
                <div className="row">
                  <div className="col-6">
                    <input
                      type="text"
                      className="form-control"
                      placeholder="cupone code"
                    />
                  </div>
                  <div className="col-6">
                    <input
                      type="submit"
                      className="btn btn-default"
                      defaultValue="Use cupone"
                    />
                  </div>
                </div>
              </div>
              <div className="float-right" style={{ margin: "10px" }}>
                <a className="btn btn-success float-right">Checkout</a>
                <div
                  className="float-right"
                  style={{ margin: "5px" }}
                  id="total-price"
                >
                  Total price:
                </div>
              </div>
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

ShoppingItems.propTypes = {
  getItems: PropTypes.func.isRequired,
  item: PropTypes.object.isRequired,
  cart: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  item: state.item,
  cart: state.cart
});

export default connect(
  mapStateToProps,
  { getItems, deleteItem, addToCart }
)(ShoppingItems);
