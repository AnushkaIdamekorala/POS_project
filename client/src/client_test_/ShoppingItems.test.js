import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ShoppingItems from "../components/ShoppingItems";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import sinon from "sinon";

import configureStore from "redux-mock-store";

const middlewares = [thunk];

Enzyme.configure({ adapter: new Adapter() });

describe("components", () => {
  describe("Login component", () => {
    const mockStore = configureStore(middlewares);
    let store;
    const loginUser = sinon.spy();
    const userCarts = sinon.spy();

    const initialState = {
      item: {
        items: []
      },
      cart: {
        cartItems: []
      }
    };

    beforeEach(() => {
      store = mockStore(initialState);
    });
    it("should render self and subcomponents", () => {
      /* const props = {
        login: jest.fn()
      }; */
      const wrapper = mount(
        <Provider store={store}>
          <ShoppingItems />
        </Provider>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
