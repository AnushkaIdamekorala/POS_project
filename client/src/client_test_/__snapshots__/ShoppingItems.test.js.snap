// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Login component should render self and subcomponents 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ShoppingItems)>
    <ShoppingItems
      addToCart={[Function]}
      cart={
        Object {
          "cartItems": Array [],
        }
      }
      deleteItem={[Function]}
      getItems={[Function]}
      item={
        Object {
          "items": Array [],
        }
      }
    >
      <div
        className="containerer"
      >
        <div
          className="card shopping-cart"
        >
          <div
            className="card-header bg-dark text-light"
          >
            <span>
               
            </span>
            Here is what we got!!!
            <div
              className="clearfix"
            />
          </div>
          <div
            className="card-body"
          />
          <div
            className="card-footer"
          >
            <div
              className="coupon col-md-5 col-sm-5 no-padding-left float-left"
            >
              <div
                className="row"
              >
                <div
                  className="col-6"
                >
                  <input
                    className="form-control"
                    placeholder="cupone code"
                    type="text"
                  />
                </div>
                <div
                  className="col-6"
                >
                  <input
                    className="btn btn-default"
                    defaultValue="Use cupone"
                    type="submit"
                  />
                </div>
              </div>
            </div>
            <div
              className="float-right"
              style={
                Object {
                  "margin": "10px",
                }
              }
            />
          </div>
        </div>
      </div>
    </ShoppingItems>
  </Connect(ShoppingItems)>
</Provider>
`;
